exasol/analytics/schema/statements.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from typing import Any
+
+from exasol.analytics.schema.column import Column
+from exasol.analytics.schema.column_name import ColumnName
+from exasol.analytics.schema.values import quote_value
+
+
+class InsertStatement:
+    """
+    Columns and values for creating an SQL INSERT statement.
+    """
+
+    def __init__(self, columns: list[Column], separator: str = ",\n  "):
+        self._lookup = {c.name.name: c.name for c in columns}
+        self._separator = separator
+        self._columns: list[ColumnName] = []
+        self._values: list[str] = []
+
+    def add(self, values: dict[str, Any], quote_values: bool = True) -> InsertStatement:
+        """
+        Add a list of columns and values specified as dict to the
+        statement.
+        Columns are sorted by name and looked up in attribute `_lookup`.
+        Values are quoted according to parameter `quote_values` and wrt. their
+        data type.
+        """
+
+        def col_val(name: str) -> str:
+            val = values[name]
+            if val is None or quote_values:
+                return quote_value(val)
+            return str(val)
+
+        names = sorted(values)
+        self._columns += [self._lookup[n] for n in names]
+        self._values += [col_val(n) for n in names]
+        return self
+
+    def add_references(self, *references: ColumnName) -> InsertStatement:
```

Comment:
It took me a while to understand that the intention here is to create a column where the value is a full name of another column in some other table. Do we really need to provide a separate method for this? If we do, the usage should be explained better.

Reply:
Hello @ahsimb I always do hard "making something better". Could you be more specific, e.g. propose an improved docstring?

Reply:
See next push for an updated docstring.